src/types/ambient/outlinePlugin.d.ts

```diff
@@ -34,14 +37,14 @@ declare namespace cordova.plugins.outline {
 
   // Implements the Tunnel interface with native functionality.
   class Tunnel implements Tunnel {
-    // Creates a new instance with `config`.
-    // A sequential ID will be generated if `id` is absent.
-    constructor(config: ShadowsocksConfig, id?: string);
+    constructor(id: string, config?: ShadowsocksConfig);
 
-    config: ShadowsocksConfig;
+    config?: ShadowsocksConfig;
 
     readonly id: string;
 
+    fetchProxyConfig(source: ShadowsocksConfigSource): Promise<ProxyConfigResponse>;
```

Comment:
This would go away, since the TS code would already be giving you the Shadowsocks config to use.